resilience4j-retrofit/src/main/java/io/github/resilience4j/retrofit/CircuitBreakerCallAdapterFactory.java

```diff
@@ -0,0 +1,60 @@
+package io.github.resilience4j.retrofit;
+
+import io.github.resilience4j.circuitbreaker.CircuitBreaker;
+import retrofit2.Call;
+import retrofit2.CallAdapter;
+import retrofit2.Response;
+import retrofit2.Retrofit;
+
+import java.lang.annotation.Annotation;
+import java.lang.reflect.ParameterizedType;
+import java.lang.reflect.Type;
+import java.util.function.Predicate;
+
+/**
+ * Creates {@link CallAdapter} that decorates a {@link Call} to provide integration with a
+ * Javaslang {@link CircuitBreaker} using {@link RetrofitCircuitBreaker}
+ */
+public final class CircuitBreakerCallAdapterFactory extends CallAdapter.Factory {
+
+    private final CircuitBreaker circuitBreaker;
+    private final Predicate<Response> successResponse;
+
+    public CircuitBreakerCallAdapterFactory(final CircuitBreaker circuitBreaker) {
```

Comment:
We usually use factory methods all over the code.
Please add such methods too, just in order to make it more consistent with the rest of the library